src/api/app/models/package.rb

```diff
@@ -804,7 +810,7 @@ def delete_on_backend
 
     # not really packages...
     # everything below _product:
-    return if name =~ /\A_product:\w[-+\w\.]*\z/ && master_product_object != self
+    return if belongs_to_product? && project.packages.product.exists?
```

Comment:
`&& project.packages.product.exists?` is already included in `belongs_to_product?` method. No need to be included here, again.